packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -3620,7 +3623,7 @@ class InputDecoration {
       floatingLabelAlignment: floatingLabelAlignment ?? theme.floatingLabelAlignment,
       isDense: isDense ?? theme.isDense,
       contentPadding: contentPadding ?? theme.contentPadding,
-      isCollapsed: isCollapsed,
+      isCollapsed: theme.isCollapsed || isCollapsed,
```

Comment:
Should we change the type of isCollapsed in InputDecoration to be nullable?  That way we could follow the same pattern as the other fields here.

I'm assuming that InputDecorationTheme.isCollapsed is not used at all right now?

Reply:
It looks like isCollapsed was added before NNBD for both InputDecoration and InputDecorationTheme, so it might have been made non-nullable accidentally at that time.

Reply:
even though `InputDecorationTheme.isCollapsed` might not be used very often, having support for either way application would be important.

So we'll make `InputDecoration.isCollapsed` nullable, right?

Reply:
Yes, looks good the way you did it :+1: